Src/FluentAssertions/Events/EventMonitorOptions.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+
+namespace FluentAssertions.Events;
+
+/// <summary>
+/// Settings for the EventMonitor.
+/// </summary>
+public class EventMonitorOptions
+{
+    /// <summary>
+    /// Will ignore the events, if they throw an exception on any custom event accessor implementation. default: false.
+    /// </summary>
+    internal bool ShouldIgnoreEventAccessorExceptions { get; private set; }
+
+    /// <summary>
+    /// This will record the event, even if the event accessor add event threw an exception. To ignore exceptions in the event add accessor, call <see cref="IgnoreEventAccessorExceptions"/> property to set it to true. default: false.
+    /// </summary>
+    internal bool ShouldRecordEventsWithBrokenAccessor { get; private set; }
+
+    /// <summary>
+    /// Func used to generate the timestamp.
+    /// </summary>
+    internal Func<DateTime> TimestampProvider { get; private set; } = () => DateTime.UtcNow;
+
+    /// <summary>
+    /// When called it will ignore event accessor Exceptions.
+    /// </summary>
+    /// <returns>The options instance for method stacking.</returns>
+    public EventMonitorOptions IgnoreEventAccessorExceptions()
```

Comment:
🔧 Similar to `SelfReferenceEquivalencyOptions`, we should `IgnoringEventAccessorExceptions`. Same for `RecordEventsWithBrokenAccessor`